### ISSUE TITLE
Enable parallel threads to be executed even when the queue is active

### ISSUE THREAD

**Issue by apolinario** [gradio-app/gradio#1864]
- [x] I have searched to see if a similar issue already exists.

**Is your feature request related to a problem? Please describe.**  
Currently, if `enable_queue` is `True`, the amount `max_threads` gets ignored - which I agree should happen - and there is no way to run tasks in parallel - which I think should change, because, it is not always the case that having a queue up means you don't want parallelization anymore. You may still want to parallelize a certain amount of tasks before filling in the queue

**Describe the solution you'd like**  
Create a way for the user to decide how many parallel threads they want used. So if, say, `max_threads_queue == 2` , the first two requests are dealt with in parallel and only the 3rd request enters the queue, so we can use the more powerful machines in parallel instead of just one task at a time that may leave resources idle. 
If it makes sense it could become part of https://github.com/gradio-app/gradio/pull/1489 or become a new task if doesn't 

**Additional context**  
Internal discussion: https://huggingface.slack.com/archives/C02QZLG8GMN/p1658514604265339

PS: This is not the same as https://github.com/gradio-app/gradio/issues/1597, as this is CPU parallelization, although this could in part solve that problem too as you can use parallel CPU jobs to run multiple stuff on GPU as long there is VRAM to do so (just not as efficiently than a GPU-only solution)

**Comment by FarukOzderim** [comment]
Hmm, it feels like the new queue solves this issue like [this](https://github.com/gradio-app/gradio/pull/1978/files#diff-1470d66b816ce9346baf5cbabfe71f5cb3db2cfd8c34b54d3fa58405feb2c4d8R16), closing the issue, feel free to reopen.
